api-2.2/src/main/java/org/openmrs/htmlformentryui/ConditionElement.java

```diff
@@ -0,0 +1,309 @@
+			try {
+				
+				int conceptId = Integer.parseInt((String) conditionName.getValue(session.getContext(), submission));
+				conditionConcept.setCoded(new Concept(conceptId));
+				
+			} catch(NumberFormatException e) {
+				String nonCodedConcept = submission.getParameter(context.getFieldName(conditionName));
+				conditionConcept.setNonCoded(nonCodedConcept);
+			}
+			condition.setCondition(conditionConcept);
+			condition.setClinicalStatus(getStatus(context, submission));
+			condition.setOnsetDate(onSetDate.getValue(context, submission));
+			if (ConditionClinicalStatus.INACTIVE == getStatus(context, submission)) {
+				condition.setEndDate(endDate.getValue(context, submission));
+			}
+			condition.setPatient(session.getPatient());
+			conditionService = Context.getConditionService();
+			conditionService.saveCondition(condition);
+		}
+	}
+
+	@Override
+	public Collection<FormSubmissionError> validateSubmission(FormEntryContext context, HttpServletRequest submission) {
+		
+		// TODO Localize error messages
+		
+		List<FormSubmissionError> ret = new ArrayList<FormSubmissionError>();
+		Date givenOnsetDate = onSetDate.getValue(context, submission);
+		String condition = submission.getParameter(context.getFieldName(conditionName));
+		
+		if (context.getMode() != Mode.VIEW) {
+			
+			if (StringUtils.isBlank(condition)) {
+				ret.add(new FormSubmissionError(context
+	                    .getFieldName(conditionName), "Condition name required"));
+			}
+			
+			if (givenOnsetDate != null) {
+				if (givenOnsetDate.after(new Date())) {
+					ret.add(new FormSubmissionError(context
+	                        .getFieldName(onSetDate), "Onset date can't be after Today"));
+				}
+			} else {
+				ret.add(new FormSubmissionError(context
+	                    .getFieldName(onSetDate), "Onset date required"));
+			}
+			
+			ConditionClinicalStatus status = getStatus(context, submission);
+			if (status != null) {
+				if (ConditionClinicalStatus.INACTIVE == status) {
+					Date givenEndDate = endDate.getValue(context, submission);
+					if (givenEndDate != null) {
+						if (givenOnsetDate != null) {
+							if (givenOnsetDate.after(givenEndDate)) {
+								ret.add(new FormSubmissionError(context
+				                        .getFieldName(endDate), "End date can't be before onset date"));
+							}
+
+						} 
+					} else {
+						ret.add(new FormSubmissionError(context
+		                        .getFieldName(endDate), "End date required"));
+					}
+				}	
+		    } else {
+		    	ret.add(new FormSubmissionError(context
+	                    .getFieldName(conditionStatusWidget), "A Condition status is required"));
+		    }
+		}
+		return ret;
+	}
+
+	@Override
+	public String generateHtml(FormEntryContext context) {
+		mss = Context.getMessageSourceService();		
+		StringBuilder ret = new StringBuilder();
+		ret.append("<h5>");
+		ret.append(mss.getMessage("coreapps.conditionui.addNewCondition"));
+		ret.append("</h5>");
```

Comment:
I don't think we want any title. Labels will be added when the tag is being used on the form.